services/brig/src/Brig/API/User.hs

```diff
@@ -1230,10 +1232,57 @@ verifyDeleteUser d = do
   for_ account $ lift . wrapHttpClient . deleteAccount
   lift . wrapClient $ Code.delete key Code.AccountDeletion
 
--- | Internal deletion without validation.  Called via @delete /i/user/:uid@, or indirectly
--- via deleting self.
--- Team owners can be deleted if the team is not orphaned, i.e. there is at least one
--- other owner left.
+-- | Check if `deleteAccount` succeeded and run it again if needed
+-- Called via @delete /i/user/:uid@.
+ensureAccountDeleted ::
+  ( MonadLogger m,
+    MonadCatch m,
+    MonadThrow m,
+    MonadIndexIO m,
+    MonadReader Env m,
+    MonadIO m,
+    MonadMask m,
+    MonadHttp m,
+    HasRequestId m,
+    MonadUnliftIO m,
+    MonadClient m,
+    MonadReader Env m
+  ) =>
+  UserId ->
+  m DeleteUserResult
+ensureAccountDeleted uid = do
+  mbAcc <- lookupAccount uid
+  case mbAcc of
+    Nothing -> pure NoUser
+    Just acc -> do
+      probs <- Data.lookupPropertyKeysAndValues uid
+
+      let accIsDeleted = accountStatus acc == Deleted
+      clients <- Data.lookupClients uid
+
+      localUid <- qualifyLocal uid
+      conCount <- countConnections localUid [minBound .. maxBound]
+      cookies <- listCookies uid []
+
+      if (not . null) probs
```

Comment:
```suggestion
      if notNull probs
```

https://hoogle.zinfra.io/file/root/.cabal/store/ghc-8.10.7/extra-1.7.9-8f69c8c825ffedcd23560ad4ef0789aad992568ab4532f16f7edce65b84795fc/share/doc/html/Data-List-Extra.html#v:notNull

(now that i'm writing this, i wonder if `notNull` should even exist...)